File: src/Kot_B_10/KotCarBMW.java
package Kot_B_10;

public class KotCarBMW extends KotCar{
    public KotCarBMW(int fuelConsumption, int carPrice, int maxSpeed) {
        super(fuelConsumption, carPrice, maxSpeed);
    }

    @Override
    public int compareTo(KotCar o) {
        return (getFuelConsumption() - o.getFuelConsumption());
    }

    @Override
    public String toString() {
        return "KotCarBMW {" +
                "fuelConsumption=" + getFuelConsumption() +
                ", carPrice=" + getCarPrice() +
                ", maxSpeed=" + getMaxSpeed() +
                '}';
    }
}

File: src/Kot_B_10/KotCarBMW.kt
package Kot_B_10;

data class KotCarBMW(
    override var fuelConsumption: Int,
    override var carPrice: Int,
    override var maxSpeed: Int
) : KotCar(fuelConsumption, carPrice, maxSpeed) {
    override fun compareTo(other: KotCar): Int {
        return (fuelConsumption - other.fuelConsumption)
    }
}
